### PR TITLE
SB-9795 :User framework update testcases added

### PR DESCRIPTION
**Pull request by iostream04**

### CODE REVIEW

File: service/test/controllers/usermanagement/UserControllerTest.java [hunk 1]
```diff
@@ -118,6 +118,64 @@ public void testSearchUserFailureWithoutFilter() {
     assertEquals(getResponseCode(result), ResponseCode.loginIdRequired.getErrorCode());
     assertTrue(getResponseStatus(result) == 400);
   }
+  @Test
+  public void testUpdateUserFrameworkSuccess() {
+    Result result =
+        performTest(
+            "/v1/user/update",
+            "PATCH",
```
Comment (amolnin) [inline service/test/controllers/usermanagement/UserControllerTest.java]: use httpmethods constant

File: service/test/controllers/usermanagement/UserControllerTest.java [hunk 2]
```diff
@@ -118,6 +118,64 @@ public void testSearchUserFailureWithoutFilter() {
     assertEquals(getResponseCode(result), ResponseCode.loginIdRequired.getErrorCode());
     assertTrue(getResponseStatus(result) == 400);
   }
+  @Test
+  public void testUpdateUserFrameworkSuccess() {
+    Result result =
+        performTest(
+            "/v1/user/update",
```
Comment (amolnin) [inline service/test/controllers/usermanagement/UserControllerTest.java]: create constant for urls

File: service/test/controllers/usermanagement/UserControllerTest.java [hunk 3]
```diff
@@ -118,6 +118,64 @@ public void testSearchUserFailureWithoutFilter() {
...
+    innerMap.put(JsonKey.PHONE, phoneNumber);
+    innerMap.put(JsonKey.COUNTRY_CODE, "+91");
+    innerMap.put(JsonKey.EMAIL, emailId);
+    request.put(JsonKey.REQUEST, innerMap);
+    return request;
+  }
+
+  private Map<String, Object> getFrameworkDetails(String frameworkId, boolean success) {
+    Map<String, Object> frameworkMap = new HashMap<>();
+    List<String> medium = new ArrayList<>();
```
Comment (amolnin) [inline service/test/controllers/usermanagement/UserControllerTest.java]: Arrays.asList - can crreate list in single call

File: service/test/controllers/usermanagement/UserControllerTest.java [hunk 4]
```diff
@@ -118,6 +118,64 @@ public void testSearchUserFailureWithoutFilter() {
...
+    Map<String, Object> request = new HashMap<>();
+    Map<String, Object> innerMap = new HashMap<>();
+    Map<String, Object> frameworkMap;
+    frameworkMap = getFrameworkDetails(frameworkId, success);
+    innerMap.put(JsonKey.USER_ID, userId);
+    innerMap.put(JsonKey.FRAMEWORK, frameworkMap);
+    innerMap.put(JsonKey.PHONE_VERIFIED, true);
+    innerMap.put(JsonKey.PHONE, phoneNumber);
+    innerMap.put(JsonKey.COUNTRY_CODE, "+91");
+    innerMap.put(JsonKey.EMAIL, emailId);
```
Comment (amolnin) [inline service/test/controllers/usermanagement/UserControllerTest.java]: without emailverified this might fail

Comment (amolnin) [reply service/test/controllers/usermanagement/UserControllerTest.java]: u cn use either phn or email

File: service/test/controllers/usermanagement/UserControllerTest.java (hunk 1 above)
Comment (iostream04) [reply service/test/controllers/usermanagement/UserControllerTest.java]: No PATCH in HttpMethods.

File: service/test/controllers/usermanagement/UserControllerTest.java (hunk 2 above)
Comment (iostream04) [reply service/test/controllers/usermanagement/UserControllerTest.java]: Done !

File: service/test/controllers/usermanagement/UserControllerTest.java (hunk 4 above)
Comment (iostream04) [reply service/test/controllers/usermanagement/UserControllerTest.java]: updated it.

**Pull request merged**